### PR TITLE
Initial keytar commit.

### PR DESCRIPTION
**Pull request by thompsonja**
Wanted to get this out for some initial comments while I'm on vacation. There's still some minor cleanup that's needed (comments, docs, etc.), but generally the gist is there. I went with the name keytar, I'll have to backronym that later...

After this commit, we can add a dockerhub webhook that points to a running instance of keytar on GKE. Every new image will fire a json payload to the instance, triggering a cluster test run to fire using the GKE resources of the project that keytar is running under. This will result in tests running on actual cloud resources and can be considered as post-submit tests. We can discuss later how to react better to these tests.

This commit includes a small front-end that displays test results, dockerfile for creating the docker image, kubernetes templates, unit tests/webdriver tests, and some (admittedly light) documentation.

@michael-berlin 
@dumbunny for python readability.

### CODE REVIEW

File: test/cluster/keytar/keytar.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  logging.getLogger().setLevel(logging.INFO)
+  parser = argparse.ArgumentParser(description='Run keytar')
+  parser.add_argument('--config_file', help='Keytar config file')
+  parser.add_argument('--key', help='Key', default=None)
+  parser.add_argument('--port', help='Port', default=8080, type=int)
+  keytar_args = parser.parse_args()
+  if keytar_args.config_file:
+    with open(keytar_args.config_file, 'r') as yaml_file:
+      yaml_config = yaml_file.read()
+  if not yaml_config:
```
Comment (dumbunny) [inline test/cluster/keytar/keytar.py]: If keytar_args.config_file is False, yaml_config is not initialized.

File: test/cluster/keytar/keytar_web_test.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  def setUpClass(cls):
+    cls.driver = create_webdriver()
+    cls.flask_process = subprocess.Popen(
+        ['./keytar.py', '--config_file=test_config.yaml',
+         '--port=8080', '--key=foo'],
+        preexec_fn=os.setsid)
+
+  @classmethod
+  def tearDownClass(cls):
+    os.killpg(cls.flask_process.pid, signal.SIGKILL)
```
Comment (dumbunny) [inline test/cluster/keytar/keytar_web_test.py]: Is SIGKILL necessary? Is SIGTERM flaky?

File: test/cluster/keytar/test_runner.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      test_file = os.path.join(repo_dir, test['file'])
+      test_name = os.path.basename(test_file)
+      logging.info('Running test %s', test_name)
+      os.chdir(os.path.dirname(test_file))
+      test_args = [
+          './%s' % test_name,
+          '-e', environment_config['application_type'], '-n', name]
+      if 'params' in test:
+        test_args += ['-t', ':'.join(
+            ['%s=%s' % (k, v) for (k, v) in test['params'].iteritems()])]
```
Comment (dumbunny) [inline test/cluster/keytar/test_runner.py]: Extraneous inner brackets.

**Comment by thompsonja** [comment]
LGTM

**Pull request merged**